### PR TITLE
[release-0.25] Sync upstream bug fixes

### PR DESCRIPTION
**Pull request by pierDipi**
Backports:
- https://github.com/knative-sandbox/eventing-kafka/pull/895

### CODE REVIEW

**Comment by matzew** [comment]
/hold

did you patch this against the openshift 0.25 ?

**Comment by matzew** [comment]
I do _not_ see an `openshift` folder on https://github.com/pierDipi/eventing-kafka/tree/cp-895-v0.25-os  :man_shrugging:

File: config/channel/resources/kafkachannel-crd.yaml [hunk 1]
```diff
@@ -144,8 +144,6 @@ spec:
             status:
               description: Status represents the current state of the KafkaChannel. This data may be out of date.
               type: object
-              required:
-                - address
               properties:
                 address:
```
Comment (matzew) [inline config/channel/resources/kafkachannel-crd.yaml:148]: I miss something here....

this change was already done to 0.25 -> https://github.com/openshift-knative/eventing-kafka/pull/344/files


Related, we also regenerated our CI manifests -> https://github.com/openshift-knative/eventing-kafka/pull/384

But `openshift` folder is missing here`

**Comment by matzew** [comment]
For the `[Channel] Set defaults of spec.delivery knative-sandbox/eventing-kafka#895` PR

can we use the upstream merge sha?

**Comment by matzew** [comment]
```

level=error msg=Error: Error creating VPC Endpoint: InvalidRouteTableId.NotFound: The Route Table Id 'rtb-081647025e9a36255' does not exist.
level=error msg=	status code: 400, request id: 953faf6a-6106-4085-8daf-10e188f3c5a7
level=error
level=error msg=  on ../tmp/openshift-install-429540405/vpc/vpc.tf line 21, in resource "aws_vpc_endpoint" "s3":
level=error msg=  21: resource "aws_vpc_endpoint" "s3" {
level=error
```

/retest

**Comment by matzew** [comment]
@pierDipi we have touched the upstream yamls.  

We do generate our own release manifests, to be applied for CI run, and store them in `openshift/release` folder.


Can you run:
```
 make RELEASE=v0.25.3 generate-release
```

to update the files? Doing so will add the `delete` verb from the ClusterRole to the generated files for the channels

Thanks

**Comment by matzew** [comment]
Why do we have so many (unrelated) commits on this PR? :thinking:

**Comment by pierDipi** [comment]
On Tue, Nov 16, 2021 at 9:34 AM Matthias Wessendorf
***@***.***> wrote:
>
> Why do we have so many (unrelated) commits on this PR? 🤔
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

I think it's the way I created this patch, I'll update the branch
to properly include upstream commits that are missing here.

Can we merge upstream PR:
https://github.com/knative-sandbox/eventing-kafka/pull/989/files?

-- 

Pierangelo Di Pilato

Software Engineer

Red Hat, Inc

https://www.redhat.com/

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**